fix: add CVSSv4 support to auditjs parser and improve error handling

Haralishev77 opened the pull request:
### Summary

This PR adds **CVSS v4 support** to the AuditJS parser in DefectDojo.

### Changes

* Updated the parser logic to handle `CVSS:4.0` vectors using the `CVSS4` class.
* Added clear error handling when no CVSS vectors are found.
* Ensured backward compatibility with CVSS2 and CVSS3.

### Note

Support for CVSS4 relies on [RedHatProductSecurity/cvss#75](https://github.com/RedHatProductSecurity/cvss/pull/75) being merged and released (likely in version `3.5`).
Once it's available, the `requirements.txt` should be updated accordingly.

valentijnscholten commented on dojo/tools/auditjs/parser.py:
```diff
@@ -96,22 +96,32 @@ def get_findings(self, filename, test):
+                            vector_obj = cvss_vectors[0]
+
+                            if isinstance(vector_obj, CVSS4):
+                                severity = vector_obj.severities()[0]
```
Should the vector be added to the description here as well?

Haralishev77 replied:
Yes, that would make sense. Thanks!

valentijnscholten commented:
Can you look at my comment, the linting errors and rebase against the `bugfxi` branch?

valentijnscholten commented:
Thanks, I've updated the base branch here on the GitHub PR as well.

Could you look at providing/updating a test case that covers the new CVSS code?

Haralishev77 commented:
Sure, I’ll take a look at the test file next week and may extend it with additional scan examples to cover CVSS v2 and v4 cases.

Haralishev77 commented:
I've added new tests for CVSS V2 and V4 vectors and updated the scan examples. However, to run the tests properly, it's necessary to manually add the updated parse_cvss_from_text function to the auditjs parser.

valentijnscholten commented:
Thanks! I've approved the workflows to run, can you look at the test failure: https://github.com/DefectDojo/django-DefectDojo/actions/runs/15003307072/job/42158252920?pr=12391#step:8:3482

Haralishev77 commented:
Yes, this error occurs due to missing functionality in the CVSS library. So, we're waiting for updates.

```
uwsgi-1  |   File "/app/dojo/tools/auditjs/parser.py", line 119, in get_findings
uwsgi-1  |     raise ValueError(msg)
uwsgi-1  | ValueError: No CVSS vectors found. Please check that parse_cvss_from_text() correctly parses the provided cvssVector.
```

valentijnscholten reviewed (commented):
@Haralishev77 I think it would be fine if you add some extra code to handle the CVSS4 case, i.e. mimic what your upstream PR does. I am saying this because it looks like it might take a while before RedHat will release a version with your fix/enhancement. And soon we will add CVSS4 support to Defect Dojo and that will be a point where we will double check all CVSS related code in parsers and could cleanup the extra code.

Haralishev77 commented:
Okay, I’ll push the changes soon.

valentijnscholten commented on dojo/tools/auditjs/parser.py:
```diff
@@ -2,12 +2,50 @@
 from dojo.models import Finding
 
 
+# TEMPORARY: Local implementation until the upstream PR is merged
```
```suggestion
# TEMPORARY: Local implementation until the upstream PR is merged & release: https://github.com/RedHatProductSecurity/cvss/pull/75
```

valentijnscholten commented on dojo/tools/auditjs/parser.py:
```diff
@@ -2,12 +2,50 @@
 from dojo.models import Finding
 
 
+# TEMPORARY: Local implementation until the upstream PR is merged & release: https://github.com/RedHatProductSecurity/cvss/pull/75
```
```suggestion
# TEMPORARY: Local implementation until the upstream PR is merged & released: https://github.com/RedHatProductSecurity/cvss/pull/75
```

valentijnscholten reviewed (approved):
Looks good to me, feel free to take it out of draft.

mtesauro reviewed (approved):
Approved
